app/routes/auth/denied.js

```diff
@@ -1,4 +1,50 @@
 import Ember from 'ember'
 
+const NO_REASON = 6
+
 export default Ember.Route.extend({
+  queryParams: {
+    reason: {
+      refreshModel: true
+    }
+  },
+  model: function (params) {
+    if (params.reason) {
+      return params.reason
+    }
+    return NO_REASON
+  },
+  setupController: function (controller, model) {
+    controller.set('reason', model)
+    switch (model) {
```

Comment:
It seems like a weird design that the client needs to know this stuff. The server should send back some JSON to the client indicating that the attempt was successful or not and if not what the additional information is about the failure. Having to code this in both the server and the client is Too Tightly Coupled.